### ISSUE TITLE
mkdir: cannot create directory ‘/var/lib/rancher’: Permission denied

### ISSUE THREAD

**Issue by SvenDowideit** [xunleii/terraform-module-k3s#42]
I'm not that familiar with terraform, so I'm wondering if there's something simple that I do to work around not being able to ssh to root (the user I'm using to talk to this host has sudo, but needs a password - which is sadly "normal" for corp managed hosts)

```
module.k3s.module.k3s.null_resource.k8s_ca_certificates_install[5] (remote-exec): Connected!
module.k3s.module.k3s.null_resource.k8s_ca_certificates_install[1] (remote-exec): mkdir: cannot create directory ‘/var/lib/rancher’: Permission denied



module.k3s.module.k3s.null_resource.k8s_ca_certificates_install[0] (remote-exec): mkdir: cannot create directory ‘/var/lib/rancher’: Permission denied
module.k3s.module.k3s.null_resource.k8s_ca_certificates_install[2] (remote-exec): mkdir: cannot create directory ‘/var/lib/rancher’: Permission denied

Error: error executing "/tmp/terraform_1310554883.sh": Process exited with status 1
```

**Comment by xunleii** [comment]
Hi @SvenDowideit.

Firstly, thanks for using this module :smile: 

Unfortunately, I don't see any way to do that easily. How your machines are provided (cloud instances ? bare metal ?) ? Theses hosts are fully managed by IaaS (like Terraform or Ansible) ?

Currently, Terraform doesn't offer a way to sudo your current user like Ansible did. But, instead, I think it possible to create required folders with your user group rights during provisioning (through a cloud-init for example). Another possibility, but **really unsafe**, is to run a `null-resource` with a provisionner to execute a `mkdir -p` + `chmown user:group` on those folders before calling this module.

**Comment by fredleger** [comment]
we tested it a lot yesterday and it seems to work like a charm ! thanks @caleb-devops !

**Comment by xunleii** [comment]
Resolved by #49
